IoTBrowser/IoTBrowser/MainPage.xaml

```diff
@@ -36,21 +36,33 @@
         </Grid.RowDefinitions>
         
         <!--Address bar-->
-        <StackPanel Grid.Row="0" Orientation="Horizontal">
-            <TextBox x:Name="Web_Address" FontSize="24" TextWrapping="Wrap" Text="http://www.bing.com" VerticalAlignment="Center" VerticalContentAlignment="Center" Height="54" Width="958" KeyUp="Web_Address_KeyUp"/>
-            <Button x:Name="Go_Web" Content="Go!" HorizontalAlignment="Right" VerticalAlignment="Center" Height="60" Width="107" Click="Go_Web_Click"/>
-        </StackPanel>
+        <Grid Grid.Row="0" >
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="*"></ColumnDefinition>
+                <ColumnDefinition Width="10"></ColumnDefinition>
+                <ColumnDefinition Width="100"></ColumnDefinition>
+            </Grid.ColumnDefinitions>
+            <TextBox Grid.Column="0" x:Name="Web_Address" FontSize="24" TextWrapping="Wrap" Text="http://www.bing.com" VerticalAlignment="Center" VerticalContentAlignment="Center" KeyUp="Web_Address_KeyUp"/>
+            <Button Grid.Column="2" x:Name="Go_Web" Content="Go!" HorizontalAlignment="Right" VerticalAlignment="Center" Height="60" Width="100" Click="Go_Web_Click"/>
+        </Grid>
         
         <!--Web view control-->
         <WebView x:Name="webView" Grid.Row="1"/>
         
         <!--Favorites buttons-->
-        <StackPanel Grid.Row="2" Orientation="Horizontal" >
-            <Button x:Name="Go_WOD" VerticalAlignment="Center" HorizontalAlignment="Center" Height="60" Width="120" Margin="0,0,15,0" Click="Go_WOD_Click">
+        <Grid Grid.Row="2" HorizontalAlignment="Center">
```

Comment:
The buttons will fit in anything bigger than 400 px, smaller than that and the usefulness of this sample is pretty much shot anyway.